org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -410,6 +410,30 @@ public static <T> CompletableFuture<List<T>> addAll(CompletableFuture<List<T>> a
 		return res;
 	}
 
+	/**
+	 * Combines two async lists of results into a single list by adding all the elements of the second list to the first one,
+	 * ignoring abnormal results and only including succeses.
+	 *
+	 * @param <T> Result type
+	 * @param futures Async results
+	 * @return Async combined result
+	 */
+	public static <T> CompletableFuture<List<T>> addAllSuccessful(CompletableFuture<List<T>>... futures) {
+		CompletableFuture<List<T>> res = CompletableFuture.allOf(futures)
+				.thenApply( v -> {
+					List<T> results = new ArrayList<>(futures.length);
+					for(CompletableFuture<List<T>> f : futures) {
+						if (!f.isCompletedExceptionally()) {
```

Comment:
Instead of just ignoring exceptions, I think we should log them using `LanguageServerPlugin.logError`

Reply:
I disagree with logging these as errors, as I don't think this is an error in the plugin. Instead, this is handling and acceptable response from the LSP more correctly. The "errors" that are coming back are effectively saying "no matching items found". This seems like something the plugin needs to handle and not clutter the log with error messages.

Reply:
I agree the error did not happen in the client. If not logging them as an error, how do you propose that the user can realize that he did not get some results because the server returned errors?

In your particular example, I would expect that the server does not come back with an exception saying "no matching items found", but rather an empty list or null.